Library/ViewModels/PledgeViewModel.swift

```diff
@@ -449,87 +446,115 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
       updateBackingEvent.filter { $0.isTerminating }.mapConst(true)
     )
 
-    // MARK: - Success/Failure Create
+    let submitButtonTapped = Signal.merge(
+      self.submitButtonTappedSignal.mapConst(true),
+      Signal.merge(willUpdateApplePayBacking, willCreateApplePayBacking).mapConst(false)
+    )
 
-    let createPaymentAuthorizationDidFinishSignal = willCreateApplePayBacking
-      .takeWhen(self.paymentAuthorizationDidFinishSignal)
+    // MARK: - Success/Failure
 
-    let createApplePayBackingCompleted = Signal.combineLatest(
-      createApplePayBackingEvent
-        .filter { $0.isTerminating }
-        .ignoreValues(),
-      createPaymentAuthorizationDidFinishSignal
-    )
+    let scaFlowCompletedWithError = self.scaFlowCompletedWithResultSignal
+      .filter { $0.0.status == .failed }
+      .map(second)
+      .skipNil()
 
-    let createApplePayBackingEventSuccess = createApplePayBackingEvent.values()
+    let scaFlowCompletedWithSuccess = self.scaFlowCompletedWithResultSignal
+      .filter { $0.0.status == .succeeded }
+      .map(first)
+      .ignoreValues()
 
-    let createApplePayBackingEventErrors = createApplePayBackingEvent.errors()
-      .map { $0.localizedDescription }
+    let didInitiateApplePayBacking = Signal.merge(
+      willCreateApplePayBacking,
+      willUpdateApplePayBacking
+    )
 
-    let createApplePayBackingError = createApplePayBackingEventErrors
-      .takeWhen(createApplePayBackingCompleted)
+    let paymentAuthorizationDidFinish = didInitiateApplePayBacking
+      .takeWhen(self.paymentAuthorizationDidFinishSignal)
 
-    self.createBackingError = Signal.merge(
-      createApplePayBackingError,
-      createBackingEventError
+    let createOrUpdateApplePayBackingCompleted = Signal.combineLatest(
+      Signal.merge(
+        updateBackingEvent.filter { $0.isTerminating }.ignoreValues(),
+        createBackingEvent.filter { $0.isTerminating }.ignoreValues()
```

Comment:
Should this be `deprecatedCreateApplePayBackingEvent`? 🤔

Reply:
That is handled on its own so that it'll be easy to remove later. This future-proofs for when `CreateBacking` supports Apple Pay.